src/sql/src/func.rs

```diff
@@ -2461,6 +2471,38 @@ lazy_static! {
             "dense_rank" => ScalarWindow {
                 params!() => ScalarWindowFunc::DenseRank, 3102;
             },
+            "lag" => ValueWindow {
+                // All args are encoded into a single record to be handled later
```

Comment:
I'm following here the same approach used in other basic aggregate functions with multiple arguments (e.g. `string_agg`, `jsonb_object_agg`, etc), which is to encode all of them down into a single record.